src/UI/Components/Diagram/actions.ts

```diff
@@ -135,30 +154,76 @@ export function appendInstance(
       graph,
       paper,
       instanceAsTable,
-      ServiceInstanceModel,
+      serviceInstanceModel,
       serviceInstance.candidate_attributes,
       "candidate",
+      instanceToConnectRelation,
     );
   } else {
     handleAttributes(
       graph,
       paper,
       instanceAsTable,
-      ServiceInstanceModel,
+      serviceInstanceModel,
       serviceInstance.active_attributes as InstanceAttributeModel,
       "active",
+      instanceToConnectRelation,
     );
   }
-  const appendedInstances = serviceWithReferences.relatedInstances.map(
-    (relatedInstance) => {
-      return appendInstance(paper, graph, relatedInstance, services, false, {
-        id: serviceWithReferences.instance.data.id,
-        name: serviceWithReferences.instance.data.service_entity,
-      });
-    },
-  );
-  connectEntities(graph, instanceAsTable, appendedInstances);
+  serviceWithReferences.relatedInstances.forEach((relatedInstance) => {
+    const isMainInstanceBoolean = false;
+    const cellAdded = graph.getCell(relatedInstance.instance.data.id);
+    if (!cellAdded) {
+      appendInstance(
+        paper,
+        graph,
+        relatedInstance,
+        services,
+        isMainInstanceBoolean,
+        instanceAsTable,
+      );
+    } else {
+      //If cell is already in the graph, we need to check if it got in its inter-service relations the one with id that corresponds with created instanceAsTable
+      let connected = false;
```

Comment:
it's common practice to have boolean values starting with "is" or "has". So it would be isConnected or hasConnections